frontend/src/Context/user.tsx

```diff
@@ -49,6 +52,13 @@ const UserProvider = ({ children }: UserProviderProps): ReactElement => {
   const noiseSuppression = window.localStorage.getItem('noiseSuppression') === 'true';
   const blur = window.localStorage.getItem('backgroundBlur') === 'true';
   const name = window.localStorage.getItem('username') ?? '';
+  const deviceManagerRef = useRef<DeviceManagerType | null>(null);
+  if (!deviceManagerRef.current) {
+    deviceManagerRef.current = createDeviceManager();
+  }
+  deviceManagerRef.current?.updateDeviceList();
```

Comment:
For the assignation this can be simplified like:
https://github.com/Vonage/vonage-video-react-app/blob/develop/frontend/src/Context/SessionProvider/session.tsx#L154

and If you need to do any extra logic then it should in a hook

Reply:
hmm I don't think we can since ActiveSpeakerTracker is a class and the device manager isn't 🤔

Reply:
You can still instantiate the ref with createDeviceManager() directly,
`updateDeviceList` should be in a hook, although maybe createDeviceManager should be invoking it if it needs to be called ? 🤔 I need to understand the flow here better